server/property-routes.ts

```diff
@@ -222,14 +253,30 @@ router.delete('/:propertyId', async (req, res, next) => {
   try {
     console.log('Delete Property Request params:', req.params);
 
-    const propertyId = Number(req.params.propertyId);
+    const userId = Number(req.user?.userId);
+    if (!userId) {
+      throw new ClientError(401, 'Authentication required');
+    }
 
+    const propertyId = Number(req.params.propertyId);
     if (!propertyId) {
       console.warn('Invalid propertyId provided:', req.params.propertyId);
-
       throw new ClientError(400, 'propertyId is required');
     }
 
+    // Verify ownership
+    const verifyOwnerSql = `
+    SELECT * FROM "properties" 
```

Comment:
Same here. This query is not necessary. Not only that, since it's not in a transaction, it's actually possible for it to fail because another client could actually add the property in between this SELECT statement and the following DELETE (or UPDATE).